### PR TITLE
Move core scheduler typedefs to libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**
This is part of a project to merge the functionality of libpe_rules and libpe_status into libcrmcommon, bringing the code up to current guidelines. This portion renames four key types.

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk @clumens @gao-yan @HideoYamauchi FYI, this PR proposes to rename pe_node_t, pe_resource_t, and pe_action_t with a pcmk prefix, and rename pe_working_set_t to pcmk_scheduler_t. If anyone doesn't like that idea, please speak up :)

**Review by nrwahl2 — APPROVED** [review]
Looks good. I could only skim the "replace old with new" commits. Should be fine since it's search and replace, but let me know if you want me to go through with a fine-toothed comb. It will be tedious.

File: include/crm/common/scheduler_types.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#endif
+
+/*!
+ * \file
+ * \brief Type aliases needed to define scheduler objects
+ * \ingroup core
+ */
+
+//! Node object (including information that may vary depending on resource)
+typedef struct pe_node_s pcmk_node_t;
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply include/crm/common/scheduler_types.h:24]: That's still up in the air. I'm not sure whether we'll have enough time to get that deep into a redesign.

File: include/crm/common/scheduler_types.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2004-2023 the Pacemaker project contributors
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply include/crm/common/scheduler_types.h:2]: Copy-paste :)

**Comment by kgaillot** [comment]
Rebased on current main, only change is copyright year (no need to re-review)

**Review by nrwahl2 — APPROVED** [review]

**Comment by wenningerk** [comment]
> @wenningerk @clumens @gao-yan @HideoYamauchi FYI, this PR proposes to rename pe_node_t, pe_resource_t, and pe_action_t with a pcmk prefix, and rename pe_working_set_t to pcmk_scheduler_t. If anyone doesn't like that idea, please speak up :)

With pe_working_set_t renamed alone without changing content sbd-binaries would stay binary compatible but sbd wouldn't build against pacemaker anymore - right? I should be able to use AC_CHECK_TYPE (similar as with introduction of pe_new_working_set & AC_CHECK_FUNCS) to typedef the new name as to make it still compile against old pacemaker-sources.

**Comment by kgaillot** [comment]
> > @wenningerk @clumens @gao-yan @HideoYamauchi FYI, this PR proposes to rename pe_node_t, pe_resource_t, and pe_action_t with a pcmk prefix, and rename pe_working_set_t to pcmk_scheduler_t. If anyone doesn't like that idea, please speak up :)
> 
> With pe_working_set_t renamed alone without changing content sbd-binaries would stay binary compatible but sbd wouldn't build against pacemaker anymore - right? I should be able to use AC_CHECK_TYPE (similar as with introduction of pe_new_working_set & AC_CHECK_FUNCS) to typedef the new name as to make it still compile against old pacemaker-sources.

The old names would remain available, but deprecated. Pacemaker code won't be able to use them but external code can.

**Comment by kgaillot** [comment]
> > > @wenningerk @clumens @gao-yan @HideoYamauchi FYI, this PR proposes to rename pe_node_t, pe_resource_t, and pe_action_t with a pcmk prefix, and rename pe_working_set_t to pcmk_scheduler_t. If anyone doesn't like that idea, please speak up :)
> > 
> > 
> > With pe_working_set_t renamed alone without changing content sbd-binaries would stay binary compatible but sbd wouldn't build against pacemaker anymore - right? I should be able to use AC_CHECK_TYPE (similar as with introduction of pe_new_working_set & AC_CHECK_FUNCS) to typedef the new name as to make it still compile against old pacemaker-sources.
> 
> The old names would remain available, but deprecated. Pacemaker code won't be able to use them but external code can.

I wouldn't worry about the downside of using deprecated APIs until the changes in Pacemaker 3.0 are decided. At that point you should be able to test for one new symbol and define a bunch of constants for alternatives. I'm hoping the 3.0 APIs will be stable for a long time.

**Pull request merged**